### ISSUE TITLE
Type error emitted for valid type index lookups in variadic tuple types

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/TypeScript#40030]
**TypeScript Version:**  4.0.0-beta

**Search Terms:** cannot be used to index type tuple partial generic template variadic

**Code**

```ts
type WithOneMore<T extends unknown[]> = [...T, ''];

type LengthPlusOne<T extends unknown[]> = WithOneMore<T>['length'];

type TestNowFour = LengthPlusOne<['a', 'b', 'c']>;
```

**Expected behavior:**

There should be no type checker errors. `TestNowFour` is equal to the literal `4`.

**Actual behavior:**

The following error is emitted for `WithOneMore<T>['length']`:

```
ts(2536): Type '"length"' cannot be used to index type '[...T, ""]'.
```

Interestingly, `TestNowFour` is computed to be equal to the literal `4` regardless of the error.

**Playground Link:** [Playground Link](https://www.typescriptlang.org/play?ts=4.0.0-beta#code/C4TwDgpgBA6glsAFgeQHYQLIHsBOEA8AKlBAB7ASoAmAzlAK6oDWqWA7qgNoC6AfFAF4onAHRjCAGigByadwDcAKEWhIUADKUA5kgAKAG3o00BYmQrU6jFuy59BsBCnTY8RXp2n7tSOUpXg0IQQNMAAcuwAYlj0OA6aqDqIBkYm+J4AhtJS0gBG2TIAxnK88kA)

**Related Issues:** Nothing particularly close. #5453 tracked the variadic kinds proposal and #39094 implemented them. I am keenly looking at what's coming out of the recursive conditional types mentioned in #40006 though...

**Comment by tadhgmister** [comment]
as a work around `[...T, ''] & {length: number}` works as expected ([playground](https://www.typescriptlang.org/play?ts=4.0.0-beta#code/C4TwDgpgBA6glsAFgeQHYQLIHsBOEA8AKlBAB7ASoAmAzlAK6oDWqWA7qgNoC6AfFAF4onAHRjCAGigByadygAyKAG8ANpQDmSAFxRU9ALYAjCDgC+AbgBQV0JCgAZTUgAKq+jTQFiZCtTqMLOxcfIKwCCjo2HhEvJzS6qhaiHLWtuDQhBA0wABy7ABiWPQ4YU5Jru6e6PjxAIbSUtJGjTIAxnK8FkA))

**Comment by ahejlsberg** [comment]
No longer reproduces on 4.0.2 or master. I believe this was fixed before we shipped 4.0.
